### ISSUE TITLE
Feature request: write metadata.yml as preproc netcdfs are written, not at the end. 

### ISSUE THREAD

**Issue by ledm** [ESMValGroup/ESMValCore#2173]
At the moment, the metadata.yml files are written once at the end after the preproc variable group has completed.

This means that "if" ("when" in my case!) a dataset fails, the entire variable groups needs to re deleted and restarted. The actual error message is:
```
FileNotFoundError: [Errno 2] No such file or directory: '/gws/nopw/j04/ukesm/ldemora/ESMValTool_output/recipe_?/preproc/diag/variable_group/metadata.yml'
```

This can throw away a lot of CPU time, if you're doing something complex, expensive or with lots of ensemble members. If the metadata.yml were written as the preprocessed netcdf files are generated, then the `--resume_from` field would be a lot more useful and efficient. 

The solution here would be to move the write metadata function as an append, and run it after the preproc write was completed. I imagine we'd need to be careful to make sure that we don't have two writes happenning at the same time. (Perhaps write an initial metadata.yml for each netcdf file, then merge them at the end?

**Comment by bouweandela** [comment]
I agree it would be nice to improve the `resume-from` option. Unfortunately, I do not have funding to work on this at the moment, but if this would change at some point I would be happy to help out.
